src/js/tools/metroconfig.ts

```diff
@@ -64,6 +96,38 @@ function loadExpoMetroConfigModule(): {
   }
 }
 
+/**
+ * Adds Sentry Babel transformer to the Metro config.
+ */
+export function withSentryBabelTransformer(config: MetroConfig): MetroConfig {
+  const defaultBabelTransformerPath = config.transformer && config.transformer.babelTransformerPath;
+  logger.debug('Default Babel transformer path from `config.transformer`:', defaultBabelTransformerPath);
+
+  if (!defaultBabelTransformerPath) {
+    // This has to be console.warn because the options is enabled but won't be used
+    // eslint-disable-next-line no-console
+    console.warn('`transformer.babelTransformerPath` is undefined.');
+    // eslint-disable-next-line no-console
+    console.warn('Sentry Babel transformer cannot be used. Not adding it...');
+    return config;
+  }
+
+  if (defaultBabelTransformerPath) {
+    saveDefaultBabelTransformerPath(defaultBabelTransformerPath);
+    process.on('exit', () => {
+      cleanDefaultBabelTransformerPath();
+    });
+  }
+
+  return {
+    ...config,
+    transformer: {
+      ...config.transformer,
+      babelTransformerPath: require.resolve('./sentryBabelTransformer'),
```

Comment:
I don't know about the state of ESM/CJS in react native but `require.resolve` will not work in ESM.

Reply:
The `@sentry/react-native/metro` is CJS, same as the `metro` package.